### PR TITLE
Add diligent tools 252

### PR DESCRIPTION
**Pull request by AndreyMlashkin**
Specify library name and version:  **diligent-tools/2.5.2**

This is also a good place to share with all of us **why you are submitting this PR** (specially if it is a new addition to ConanCenter): is it a dependency of other libraries you want to package? Are you the author of the library? Thanks!

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by AndreyMlashkin** [comment]
@SSE4 @uilianries
I need some help here. How can I reference include files from diligent-core in DiligentTools/RenderStateNotation/CMakeLists.txt

**Comment by AndreyMlashkin** [comment]
Another thing is: there are a way too many places, where I should replace '../../../DiligentCore' to 'DiligentCore'
Maybe you know any easy way to handle this? Maybe adding a fake include path in DiligentCore?

**Comment by SSE4** [comment]
yes, it could be possible to create a dummy include hierarchy for `DiligentCore`, something like:
```
include -> dummy1 -> dummy2 -> empty file
```
the add them like:
```
self.cpp_info.includedirs.append(os.path.join(self.package_folder, "include"))
self.cpp_info.includedirs.append(os.path.join(self.package_folder, "include", "dummy1", "dummy2"))
```

**Comment by AndreyMlashkin** [comment]
@uilianries @SSE4 @jgsogo @SpaceIm @madebr @lasote @prince-chrismc @toge
please review

File: recipes/diligent-tools/all/conandata.yml [hunk 1]
```diff
@@ -1,16 +1,31 @@
...
+  "2.5.2":
+    - patch_file: "patches/0006-2.5.2-use-imgui-from-conan.diff"
+      base_path: "source_subfolder"
+    - patch_file: "patches/0007-2.5.2-use-diligent-core-dependencies.diff"
+      base_path: "source_subfolder"
+    - patch_file: "patches/0008-2.5.2-use-conan-dependencies.diff"
+      base_path: "source_subfolder"
+    - patch_file: "patches/0009-2.5.2-fix-include-path.diff"
+      base_path: "source_subfolder"
+    - patch_file: "patches/0010-fix-render-state-notation-paths.diff"
```
Comment (toge) [inline recipes/diligent-tools/all/conandata.yml:18]: For clarity, could you please add the version to the file name?

File: recipes/diligent-tools/all/conanfile.py [hunk 2]
```diff
@@ -58,12 +58,17 @@ def validate(self):
...
+            self.requires("diligent-core/2.5.2")
+            self.requires("imgui/1.85")
+        else:
+            self.requires("diligent-core/2.5.1")
+            self.requires("imgui/1.87")
         self.requires("libjpeg/9d")
         self.requires("libpng/1.6.37")
         self.requires("libtiff/4.3.0")
         self.requires("zlib/1.2.12")
+        self.requires('taywee-args/6.3.0')
```
[the remark this answers is not in the document]
Comment (AndreyMlashkin) [reply recipes/diligent-tools/all/conanfile.py:70]: It is needed in RenderStatePackager/CMakeLists.txt:

```
target_include_directories(Diligent-RenderStatePackager
	PRIVATE
    	include
        ${DILIGENT_ARGS_DIR}
)
```

and then 
```
RenderStatePackager/src/main.cpp: #include "args.hxx"
```

[the remark this answers is not in the document]
Comment (AndreyMlashkin) [reply recipes/diligent-tools/all/conanfile.py]: I have removed it for previous version, as it was added only in 2.5.2

File: recipes/diligent-tools/all/conanfile.py [hunk 3]
```diff
@@ -58,7 +58,11 @@ def validate(self):
             raise ConanInvalidConfiguration("Can't build diligent tools as shared lib")
 
     def requirements(self):
-        self.requires("diligent-core/2.5.1")
+        if self.version == "2.5.2":
+            self.requires("diligent-core/2.5.2")
+            self.requires('taywee-args/6.3.0')
+        else:
+            self.requires("diligent-core/2.5.1")
```
Comment (madebr) [inline recipes/diligent-tools/all/conanfile.py:65]: I don't think this is written with future diligent-tools releases in mind.
I would do it as follows:
```python
        self.requires(f"diligent-core/{self.version}")  # Require the same diligent-core version
        if self.version >= "2.5.2":
            self.requires('taywee-args/6.3.0')
```

or

```python
        diligent_core_version = {
            "2.5.1": "2.5.1",
            "2.5.2": "2.5.2",
        }.get(str(self.version), "2.5.2")  # require diligent-core/2.5.2 by default
        self.requires(f"diligent-core/{diligent_core_version}")
        if self.version >= "2.5.2":
            self.requires('taywee-args/6.3.0')
```

Comment (AndreyMlashkin) [reply recipes/diligent-tools/all/conanfile.py]: done.
But there is no diligent-tools/2.5.1 tag, no tags were created back then

**Review by madebr — APPROVED** [review]

**Comment by AndreyMlashkin** [comment]
@uilianries @SSE4 @jgsogo @SpaceIm @madebr @lasote @prince-chrismc @toge
please review

**Review by SSE4 — APPROVED** [review]

File: recipes/diligent-tools/all/conanfile.py [hunk 4]
```diff
@@ -58,8 +58,14 @@ def validate(self):
...
     def requirements(self):
-        self.requires("diligent-core/2.5.1")
-        self.requires("imgui/1.87")
+        if self.version >= "2.5.2":
+            self.requires("diligent-core/2.5.2")
+            self.requires('taywee-args/6.3.0')
+            self.requires("imgui/1.85")
+        else:
+            self.requires("diligent-core/2.5.1")
+            self.requires("imgui/1.87")
```
[the remark this answers is not in the document]
Comment (AndreyMlashkin) [reply recipes/diligent-tools/all/conanfile.py:67]: There is no way to have a common logic here for now. I can't prognoze what will be with diligent/2.5.3 and if upstream will create releases regulary. And if upstream will bump imgui version when new tag is created.

> are you sure that imgui/1.87 doesn't work for all versions packaged here?
Yes, look at build results before commit ead7719 "downgrade imgui to make macos happy"

Witch such unsertancy, I would not build some complex logic, trying predicting the future the perfect way

File: recipes/diligent-tools/all/conanfile.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (toge) [reply recipes/diligent-tools/all/conanfile.py]: I understand. Thank you.

**Review by toge — APPROVED** [review]

**Pull request merged**